docs/advanced/dcrtime.md

```diff
@@ -0,0 +1,105 @@
+
+1. A 32-byte hash is created for each document or piece of data to be timestamped.
+2. A single merkle root is calculated from the hashes.
+3. This merkle root is included in an on-chain transaction, "anchoring" the data in the Decred blockchain by timestamping the data.
+4. A transaction hash and merkle path is generated, which can then be used to query the Decred blockchain and retreive the block containing the transaction (and associated timestamp).
+
+
+## Dcrtime architecture
+
+The dcrtime service is comprised of four components:
+
+* **dcrtime:** Reference client application
+* **dcrtimed (proxy mode):** The dcrtimed server application in proxy mode. In proxy mode, dcrtimed forwards requests to the backend server (the same dcrtimed server application but in server mode). This is done as a security measure, so that a daemon with a wallet passphrase is not directly connected to the internet. This also allows the frontend to scale.
+* **dcrtimed (server mode):** The dcrtimed server application in server mode. In server mode, dcrtimed manages timestamps forwarded to it by the dcrtimed proxy and creates the Decred transaction that anchors transactions in the blockchain.
+* **dcrwallet:** A Decred wallet daemon.
+* **dcrd:** A full node implementation of Decred.
+
+The diagram below illustrates how the components are connected. 
+
+![dcrtime architecture](/img/dcrtime_architecture_diagram.png)
+
+## Dcrtime implementation
+
+In this section, we'll explore how the dcrtime architecture is implemented using two examples: a simple example where we timestamp a single text file from the command line, and a more complex example that uses dcrtime to episodically timestamp data from Politeia.
+
+If you wish to follow along, instructions on setting up dcrtime can be found in the [dcrtime repo](https://github.com/decred/dcrtime).
+
+### Simple example
+
+In this example, we'll use the dcrtime Command Line Interface (CLI) to timestamp a single text file, `myfile.txt`.
+
+The below command submits the file to Decred's timestamping server running on testnet.
+
+```
+$ dcrtime --testnet -v myfile.txt
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 Upload myfile.txt
+8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6 OK     myfile.txt
+Collection timestamp: 1497009600
+```
+
+As we can see from the output, dcrtime first creates a "digest" from `myfile.txt` (a base64 encoded hash of the file):
+
+`8496855341883fdc90cc532f8304d1c46a60586fb15d99f07e41bb5ab19c79c6`
+
+It also creates a "Collection timestamp" corresponding to the time the file was received by the server {SB: true?}.
+
+`Collection timestamp: 1497009600`
+
+If we immediately ask the server if the timestamp has been verified, it will show `Not anchored`. This is because our transaction containing the digest (specifically the merkle root of the digest) has not yet been mined into a block.
```

Comment:
Merkle root has nothing to do with this. You'll get `Not anchored` until it is time to generate the Merkle which then is anchored. This happens around every start of the hour.

Reply:
Changed to this:

>If we immediately ask the server if the timestamp has been verified, it will show `Not anchored`. This is because our transaction containing the digest has not yet been mined into a block.